src/ast/mod.rs

```diff
@@ -2655,7 +2655,16 @@ pub struct OnConflict {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum OnConflictAction {
     DoNothing,
-    DoUpdate(Vec<Assignment>),
+    DoUpdate(DoUpdate),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct DoUpdate {
+    /// Column assignments
+    pub assignments: Vec<Assignment>,
+    /// WHERE
+    pub selection: Option<Expr>,
```

Comment:
My preference is to call this `where` to match the SQL more closely -- I will make this change in a follow on PR

Reply:
Actually, now I see the name `selection` is consistent with the rest of this crate (e.g.  https://github.com/sqlparser-rs/sqlparser-rs/blob/faf75b71613431e66ec22c82e3c9ad6157474458/src/ast/mod.rs#L1139)

Thank you